Formula/pulseaudio.rb

```diff
@@ -57,11 +51,28 @@ class Pulseaudio < Formula
     depends_on "libcap"
 
     resource "XML::Parser" do
-      url "https://cpan.metacpan.org/authors/id/T/TO/TODDR/XML-Parser-2.44.tar.gz"
-      sha256 "1ae9d07ee9c35326b3d9aad56eae71a6730a73a116b9fe9e8a4758b7cc033216"
+      url "https://cpan.metacpan.org/authors/id/T/TO/TODDR/XML-Parser-2.46.tar.gz"
+      sha256 "d331332491c51cccfb4cb94ffc44f9cd73378e618498d4a37df9e043661c515d"
     end
   end
 
+  # Fix macOS build errors:
+  # - ld: unknown option: -z
+  # - ld: unknown option: -version-script=/private/tmp/pulseaudio-...
+  # - ld: unknown option: --no-undefined
+  # - Undefined symbols for architecture x86_64:
+  #     "_AbsoluteToNanoseconds", referenced from:
+  #         _pa_rtclock_age in pulsecore_core-rtclock.c.o
+  #         _pa_rtclock_get in pulsecore_core-rtclock.c.o
+  #         _pa_rtclock_from_wallclock in pulsecore_core-rtclock.c.o
+  #         _pa_timeval_rtstore in pulsecore_core-rtclock.c.o
+  #     "_pa_poll", referenced from:
+  #         _pa_autospawn_lock_acquire in pulsecore_lock-autospawn.c.o
+  #
+  # FIXME: test error:
+  # - E: [] ltdl-bind-now.c: Failed to open module .../module-allow-passthrough.so
+  patch :DATA
```

Comment:
It definitely shows that upstream only supports Linux. I'll probably look into dependents and see if we can drop usage on macOS since the state of `pulseaudio` is a bit questionable there.

Will try upstreaming some build fixes later.

Still need to figure out why modules are trying to load `.so` files rather than `.dylib`. Tried changing `PA_SOEXT` but it didn't work.